### PR TITLE
Extender documentation for the KeytarApi for SCS

### PR DESCRIPTION
**Pull request by JillieBeanSim**
Signed-off-by: Billie Simmons <49491949+JillieBeanSim@users.noreply.github.com>

## Proposed changes

Documentation in the Docs/Extenders directory for extenders on how to use the KeytarApi() in Zowe Explorer API.

## Release Notes

Milestone: 1.17.0

Changelog:

## Types of changes

What types of changes does your code introduce to Zowe Explorer?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Updates to Documentation or Tests (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This checklist will be used as reference for both the contributor and the reviewer_

- [x] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/vscode-extension-for-zowe/wiki/Best-Practices:-Contributor-Guidance) wiki
- [x] PR title follows [Conventional Commits Guidelines](https://www.conventionalcommits.org/en/v1.0.0-beta.2/)
- [x] PR Description is included
- [ ] gif or screenshot is included if visual changes are made
- [ ] `yarn workspace vscode-extension-for-zowe vscode:prepublish` has been executed
- [x] All checks have passed (DCO, Jenkins and Code Coverage)
- [ ] I have added unit test and it is passing
- [ ] I have added integration test and it is passing
- [ ] There is coverage for the code that I have added
- [ ] I have tested it manually and there are no regressions found
- [x] I have added necessary documentation (if appropriate)
- [ ] Any PR dependencies have been merged and published (if appropriate)

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by lauren-li — DISMISSED** [review]
This PR looks good to me. Thanks @JillieBeanSim for adding in this helpful guidance for extenders!

File: docs/extenders/SecureCredentials.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# Secure Credentials API Adoption
+
+Extenders may have a need for z/OS actions to be taken outside of Zowe Explorer using Zowe profiles that are managed by the Secure Credentials Store. Zowe Explorer API has a way for extenders to handle this situation.
```
Comment (IgorCATech) [inline docs/extenders/SecureCredentials.md:3]: We are starting this article with what's essentially a note. Maybe we can rearrange this somehow? 

I would start the article with a description of how to adopt the Secure Credentials API. For example,

As a Zowe Explorer developer/user, use the secure credentials API to {achive a certain goal/goals}. {Short sentence about the business value. In other words, the reason customers want/need to use this API to begin with}.

File: docs/extenders/SecureCredentials.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+The `KeytarApi()` is available in `@zowe/zowe-explorer-api` version 1.15.1 and higher. Use of this API will need to be handled before the extension is registered with Zowe Explorer. This can be done by following the steps below:
```
Comment (IgorCATech) [inline docs/extenders/SecureCredentials.md:5]: The first two sentences sound like prerequisites. Maybe, we should list them as pre-reqs? Then we can say something like: "Once you meet the pre-reqs, you can adopt/use the API.

File: docs/extenders/SecureCredentials.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+```typescript
+export function activate(context: vscode.ExtensionContext) {
+  const log = imperative.Logger.getAppLogger();
+  const keytarApi = new KeytarApi(log);
+  await keytarApi.activateKeytar(imperative.CredentialManagerFactory.initialized, EnvironmentManager.isTheia());
+}
+```
+
+After the `KeytarApi()` has been initialized, [registration with Zowe Explorer](../README-Extending.md#accessing-the-zowe-explorer-api) can begin.
```
Comment (IgorCATech) [inline docs/extenders/SecureCredentials.md:27]: It's not very clear (maybe just to me) why one needs to 'register with Zowe Explorer'. Maybe, we should add a bit more context

Comment (IgorCATech) [reply docs/extenders/SecureCredentials.md:27]: I might lack technical knowledge to understand the entire procedure, please forgive me if some of the suggestions are stupid

**Review by IgorCATech — COMMENTED** [review]
Left some comments. Let me know if they make sense.

File: docs/extenders/SecureCredentials.md (hunk 2 above)
Comment (JillieBeanSim) [reply docs/extenders/SecureCredentials.md:5]: @IgorCATech no prereqs are needed.  This doc is to help Zowe Explorer extenders to understand how the new Zowe Explorer APi  `KeytarApi()` can be used/incorporated into their VS Code extension.

**Review by lauren-li — APPROVED** [review]
These changes look good to me. Thanks @JillieBeanSim for the updates, and thanks @IgorCATech for the suggestions!

**Review by jellypuno — DISMISSED** [review]
awesome!

**Review by IgorCATech — COMMENTED** [review]
@JillieBeanSim I've pushed a commit with some changes. Please, review and let me know if they make sense :)

**Comment by JillieBeanSim** [comment]
> @JillieBeanSim I've pushed a commit with some changes. Please, review and let me know if they make sense :)

LGTM, thanks @IgorCATech

**Review by jellypuno — APPROVED** [review]

**Review by IgorCATech — APPROVED** [review]

**Pull request merged**